### PR TITLE
Ensure gasket/core is not bundled to allow GASKET_ENV

### PR DESCRIPTION
**Pull request by agerard-godaddy**
## Summary

With gasket files, being imported into app code, `@gasket/core` was being bundled. When this happens, any process.env vars are not captured, including `GASKET_ENV` which is necessary for configuring gaskets to be run in different environments.

Without this change, it would be necessary to `next build` uniquely for each environment.

## Changelog

**@gasket/plugin-nextjs**
- Add Webpack externals for `@gasket/core` in Next.js apps.

## Test Plan

N/A

### CODE REVIEW

File: packages/gasket-plugin-nextjs/lib/webpack-config.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const isGasketCore = /@gasket[/\\](core)/;
+
+function validateNoGasketCore(ctx, callback) {
```
Comment (mmason2-godaddy) [inline packages/gasket-plugin-nextjs/lib/webpack-config.js:3]: Nit: Doc block here.

**Review by mmason2-godaddy — APPROVED** [review]

File: packages/gasket-plugin-nextjs/lib/webpack-config.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+const isGasketCore = /@gasket[/\\](core)/;
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-nextjs/lib/webpack-config.js:1]:
```suggestion
const isGasketCore = /@gasket[/\\]core[$/\\]/;
```

File: packages/gasket-plugin-nextjs/lib/webpack-config.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+function validateNoGasketCore(ctx, callback) {
+  if (isGasketCore.test(ctx.request)) {
+    return callback(new Error('@gasket/core should not be used in browser code.'));
+  }
+  return callback();
+}
+
+/**
+ * Externalize @gasket/core in the server build.
+ * We do this to enable GASKET_ENV to be picked up and passed to the plugins.
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-nextjs/lib/webpack-config.js:12]: I just wonder if there's a way we can avoid using an environment variable altogether except in the case of initialization.

File: packages/gasket-plugin-nextjs/lib/webpack-config.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Configure Next.js Webpack for @gasket/core
+ *
+ * @param {Object} gasket - The gasket API.
+ * @param {Object} webpackConfig - Initial Next.js webpack config
+ * @returns {Object} Partial webpack config with UXCore2 support.
+ * @public
+ */
+function webpackConfigHook(gasket, webpackConfig) {
+  if (webpackConfig.name === 'client') {
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-nextjs/lib/webpack-config.js:36]: Is `name` the best way to detect this? Shouldn't we use the `isServer` property [from the docs](https://nextjs.org/docs/app/api-reference/next-config-js/webpack)?

File: packages/gasket-plugin-nextjs/lib/webpack-config.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    webpackConfig.externals.unshift(validateNoGasketCore);
+  } else {
+    webpackConfig.externals.unshift(externalizeGasketCore);
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-nextjs/lib/webpack-config.js:39]: A bit more succinct?

```suggestion
  webpackConfig.externals.unshift(webpackConfig.isServer ? externalizeGasketCore : validateNoGasketCore);
```

File: packages/gasket-plugin-nextjs/lib/webpack-config.js (hunk 4 above)
Comment (agerard-godaddy) [reply packages/gasket-plugin-nextjs/lib/webpack-config.js:36]: I'll try it out - we want it to externalize gasket/core from edge runtime bundles also, which is another non-client named config. If `isServer` covers that then 👍

File: packages/gasket-plugin-nextjs/lib/webpack-config.js (hunk 3 above)
Comment (agerard-godaddy) [reply packages/gasket-plugin-nextjs/lib/webpack-config.js:12]: As we move away from the CLI-driven approach, using this env var will be more necessary.

File: packages/gasket-plugin-nextjs/lib/webpack-config.js (hunk 5 above)
Comment (agerard-godaddy) [reply packages/gasket-plugin-nextjs/lib/webpack-config.js]: opting for the longer form as it is more readable and flexible, should we need to adjust the webpack config further.

**Comment by jpina1-godaddy** [comment]
There's a ts test failure

**Review by jpina1-godaddy — APPROVED** [review]

**Pull request merged**